experimental/gradually-typed-examples/Setup.hs

```diff
@@ -0,0 +1,153 @@
+import Data.List (isPrefixOf)
+import Distribution.Simple
+import Distribution.Simple.Program
+import Distribution.Simple.Setup
+import Distribution.Simple.LocalBuildInfo
+import Distribution.Types.LocalBuildInfo
+import Distribution.Types.GenericPackageDescription
+import Distribution.Types.HookedBuildInfo
+import Distribution.PackageDescription
+import Distribution.System
+import System.Directory
+import System.FilePath
+import System.Process
+import System.IO.Temp
+import Control.Monad
+import Network.HTTP.Simple
+import qualified Data.ByteString.Lazy as LBS
+import System.Environment (lookupEnv)
+import Data.Maybe (fromMaybe)
+import Control.Exception
+import GHC.IO.Exception (IOException)
+
+main :: IO ()
+main = defaultMainWithHooks $ simpleUserHooks
+  { preConf = \_ _ -> do
+      pure emptyHookedBuildInfo
+  , confHook = \(gpd, hbi) flags -> do
+      mTokenizersDir <- ensureTokenizers
+      case mTokenizersDir of
+        Nothing -> do
+          putStrLn "libtokenizers not found, skipping configuration."
+          confHook simpleUserHooks (gpd, hbi) flags
+        Just tokenizersDir -> do
+          let tokLibDir    = tokenizersDir </> "lib"
+              tokInclude   = tokenizersDir </> "include"
+          let updatedFlags = flags
+                { configExtraLibDirs     = tokLibDir : configExtraLibDirs flags
+                , configExtraIncludeDirs = tokInclude : configExtraIncludeDirs flags
+                }
+          lbi <- confHook simpleUserHooks (gpd, hbi) updatedFlags
+          return $ case buildOS of
+            OSX   -> lbi { withPrograms = addRPath tokLibDir (withPrograms lbi) }
+            Linux -> lbi { withPrograms = addRPath tokLibDir (withPrograms lbi) }
+            _     -> lbi
+  }
+
+-- === tokenizers settings ===
+tokenizersVersion :: String
+tokenizersVersion = "v0.1"
+
+getLocalUserTokenizersDir :: IO FilePath
+getLocalUserTokenizersDir = do
+  mHome <- lookupEnv "TOKENIZERS_HOME"
+  base  <- case mHome of
+    Just h  -> pure h
+    Nothing -> getXdgDirectory XdgCache "tokenizers"
```

Comment:
This will have to be `getXdgDirectory XdgCache ""`

Or else it will be written to: `~/.cache/tokenizers/libtokenizers`. The rest of the program assumes that the path is ``~/.cache/tokenizers/libtokenizers`.

Reply:
Tzhe libtokenizers are from https://github.com/hasktorch/tokenizers/tree/master/bindings/haskell/tokenizers-haskell, so I will delete the download script.